### PR TITLE
Improve XYDelaunay Boundary ID Assignment

### LINKED ISSUE #27828: Prevent XYDelaunay from Creating Default Boundary IDs That Conflict with the IDs of Its Stitched Holes.
### Motivation

If not specified, XYDelaunayGenerator assigns 0 for the external boundary, and 1,2,... for the hole boundaries. However, if the hole meshes are stitched to the generated triangulation mesh, it is possible that the hole meshes already contain these boundary IDs, which might lead to difficulties in following simulation setup.

### Design

Improve the default boundary IDs generation algorithm to avert this kind of conflicts.

### Impact

Better user experience.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #27828

### CODE REVIEW

**Review by GiudGiud — APPROVED** [review]

File: test/tests/meshgenerators/xy_delaunay_generator/tests [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [Tests]
   design = 'meshgenerators/XYDelaunayGenerator.md'
-  issues = '#20192 #26776 #27545'
+  issues = '#20192 #26776 #27545 #27828'
```
Comment (GiudGiud) [inline test/tests/meshgenerators/xy_delaunay_generator/tests:3]: we should make one of the exodiffs use pedantic to make sure this does not regress. you ll need a regold

**Comment by loganharbour** [comment]
> @GiudGiud Any idea how to solve the OpenMPI error? It seems the global ids are not matching when doing `pedantic`.

I believe there's an exodiff option to map those

**Comment by GiudGiud** [comment]
Can you disallow renumbering maybe? That can be done in the Mesh block

**Comment by miaoyinb** [comment]
It's actually missing a `comm().max()`. I think I fixed it (hopefully).

**Comment by miaoyinb** [comment]
@GiudGiud are the controlled/internal app tests failures relevant? Thanks.

**Comment by GiudGiud** [comment]
the blue crab failures are irrelevant
the sockeye failures seem related. Do you have access to the source for that?

**Comment by miaoyinb** [comment]
> the blue crab failures are irrelevant the sockeye failures seem related. Do you have access to the source for that?

Yes, I have source access for Sockeye. Let me check.

**Comment by miaoyinb** [comment]
https://github.inl.gov/ncrc/sockeye/pull/770

**Comment by GiudGiud** [comment]
@joshuahansel

**Pull request merged**